ordered_product/tests.py

```diff
@@ -0,0 +1,147 @@
+import pytest
+from ordered_product.models import Supplier, Client, Product, SupplierProduct, DeliveryLocation, OrderedProduct
+import datetime
+from django.core.exceptions import ValidationError
+
+
+@pytest.fixture
+def supplier0():
+    return Supplier(user_name="bestSup",
+                    first_name="Yuval",
+                    last_name="Cohen",
+                    password="1234",
+                    business_name="bestSup")
+
+
+@pytest.fixture
+def client0():
+    return Client(user_name="meitar1996",
+                  first_name="Meitar",
+                  last_name="Rizner",
+                  password="123456",
+                  area="North Israel")
+
+
+@pytest.fixture
+def product0():
+    return Product(qr_code="Q5o76MbdiNbXprNEnHfpcGWFp1CMF8XY",
+                   product_name="Apple",
+                   description="Sweety!")
+
+
+@pytest.fixture
+def supplier_product0(product0, supplier0):
+    return SupplierProduct(suppler_product_id=1,
+                           qr_code=product0,
+                           user_name=supplier0,
+                           price=10,
+                           quantity=50)
+
+
+@pytest.fixture
+def delivery_location0(supplier0):
+    return DeliveryLocation(user_name=supplier0,
+                            location="Kiryat Shmona",
+                            date=datetime.date(2022, 12, 30))
+
+
+@pytest.fixture
+def ordered_product0(delivery_location0, client0, supplier_product0):
+    return OrderedProduct(
```

Comment:
In here your tests pass because in your 0002 test file you have the same product,supplier,supplierproduct,deliverylocation (if you change the fields value i think all your tests will fail)

so before returning the ordered product save each one of the parameters deliver_location0....... by calling the functions that we have created in our models and after that return the OrderedProduct

Reply:
I'm using the supplier0 that Paz wrote and the tests still pass.. So I don't need to do these changes.